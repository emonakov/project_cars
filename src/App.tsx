import React, { lazy, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import ContextProvider from './components/Context';
import Header from './components/Header';
import Footer from './components/Footer';
import Fallback from './components/Fallback';
import theme from './config/theme';

const Home = lazy(() => import('./components/Page/Home'));
const CarDetails = lazy(() => import('./components/Page/CarDetails'));

const App: React.FC = () => (
    <ThemeProvider theme={theme}>
        <Router>
            <Suspense fallback={<Fallback />}>
                <Header />
                <ContextProvider>
                    <Switch>
                        <Route path="/" exact component={Home} />
                        <Route path="/car/:stockId(\d+)" exact component={CarDetails} />
                    </Switch>
                </ContextProvider>
                <Footer />
            </Suspense>
        </Router>
    </ThemeProvider>
);

export default App;
