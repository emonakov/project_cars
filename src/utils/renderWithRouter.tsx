import React from 'react';
import { Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';

const history = createMemoryHistory();

const renderWithRouter = (Component: JSX.Element, initialRoute = '/') => {
    history.push(initialRoute);

    return render(
        <Router history={history}>
            {Component}
        </Router>,
    );
};

export default renderWithRouter;
