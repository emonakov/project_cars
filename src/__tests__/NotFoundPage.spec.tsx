import React from 'react';
import { wait } from '@testing-library/react';

import App from '../App';
import renderWithRouter from '../utils/renderWithRouter';

describe('Tests 404 page', () => {
    it('Renders the 404 page when url does not match any routes', async () => {
        const { getByText, container } = renderWithRouter(<App />, '/some/nonexistent/url');
        await wait();

        expect(getByText(/404 - Not found/i)).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
