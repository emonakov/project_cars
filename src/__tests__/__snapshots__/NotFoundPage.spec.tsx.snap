// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests 404 page Renders the 404 page when url does not match any routes 1`] = `
<h4
  class="MuiTypography-root MuiTypography-h4"
>
  404 - Not found
</h4>
`;
